src/vs/workbench/contrib/terminal/browser/xterm/quickFixAddon.ts

```diff
@@ -251,42 +258,42 @@ export class TerminalQuickFixAddon extends Disposable implements ITerminalAddon,
 			e.classList.add(...quickFixSelectors);
 			updateLayout(this._configurationService, e);
 			this._audioCueService.playAudioCue(AudioCue.terminalQuickFix);
-			this._register(dom.addDisposableListener(e, dom.EventType.CLICK, () => {
-				const rect = e.getBoundingClientRect();
-				const anchor = {
-					x: rect.x,
-					y: rect.y,
-					width: rect.width,
-					height: rect.height
-				};
-				// TODO: What's documentation do? Need a vscode command?
-				const actions = fixes.map(f => new TerminalQuickFix(f, f.class || TerminalQuickFixType.Command, f.source, f.label));
-				const documentation = fixes.map(f => { return { id: f.source, title: f.label, tooltip: f.source }; });
-				const actionSet = {
-					// TODO: Documentation and actions are separate?
-					documentation,
-					allActions: actions,
-					hasAutoFix: false,
-					validActions: actions,
-					dispose: () => { }
-				} as ActionSet<TerminalQuickFix>;
+			const rect = e.getBoundingClientRect();
+			const anchor = {
+				x: rect.x,
+				y: rect.y,
+				width: rect.width,
+				height: rect.height
+			};
+			// TODO: What's documentation do? Need a vscode command?
+			const actions = fixes.map(f => new TerminalQuickFix(f, f.class || TerminalQuickFixType.Command, f.source, f.label));
+			const documentation = fixes.map(f => { return { id: f.source, title: f.label, tooltip: f.source }; });
+			const actionSet = {
+				// TODO: Documentation and actions are separate?
+				documentation,
+				allActions: actions,
+				hasAutoFix: false,
+				validActions: actions,
+				dispose: () => { }
+			} as ActionSet<TerminalQuickFix>;
 
-				const parentElement = e.parentElement?.parentElement?.parentElement?.parentElement;
-				if (!parentElement) {
-					return;
-				}
-				const delegate = {
-					onSelect: async (fix: TerminalQuickFix) => {
-						fix.action?.run();
-						this._actionWidgetService.hide();
-					},
-					onHide: () => {
-						this._terminal?.focus();
-					},
-				};
-				this._actionWidgetService.show('quickFixWidget', toMenuItems(actionSet.validActions, true), delegate, anchor, parentElement);
-			}));
+			const parentElement = e.parentElement?.parentElement?.parentElement?.parentElement;
+			if (!parentElement) {
+				return;
+			}
+			const delegate = {
+				onSelect: async (fix: TerminalQuickFix) => {
+					fix.action?.run();
+					this._actionWidgetService.hide();
+				},
+				onHide: () => {
+					this._terminal?.focus();
+				},
+			};
+			this._currentFixArgs = { items: toMenuItems(actionSet.validActions, true), delegate, anchor, parentElement };
```

Comment:
It's a little strange that this is getting created when the decoration renders, could we instead move the handling of list items into `showMenu` and use `this._quickFixes` to get it?

Reply:
we need the context from the latest render, but yeah makes sense to move the list item creation to `showMenu`

Reply:
oh I had forgotten but we set `this._quickFixes` to undefined on the next command start, so that won't work

Reply:
It would be better to store the raw fixes, say as `_previousQuickFixes`, instead of constructing the actionSet, etc. and not using them